### PR TITLE
fix: Remove Helm deployment of NICClusterPolicy

### PR DESCRIPTION
**Pull request by killianmuldoon**
Remove the ability to create the NICClusterPolicy CR through helm.values. This feature has been deprecated for a number of releases and the documentation was updated to remove reference to this feature in the most recent release.

Users should instead use the helm package to install the Network Operator and configure the system by creating the NICClusterPolicy in a separate step.

### CODE REVIEW

File: deployment/network-operator/templates/tests/test-rping.yaml [hunk 1]
```diff
@@ -1,49 +0,0 @@
-{{- if and .Values.deployCR .Values.rdmaSharedDevicePlugin .deploy .Values.secondaryNetwork.deploy .Values.secondaryNetwork.multus.deploy .Values.secondaryNetwork.cniPlugins.deploy }}
```
Comment (killianmuldoon) [inline deployment/network-operator/templates/tests/test-rping.yaml:1]: Were these tests being used anywhere? They won't work after this change, but I'm wondering if this change means some loss in testing coverage.

**Review by almaslennikov — APPROVED** [review]

**Review by e0ne — APPROVED** [review]

**Comment by rollandf** [comment]
Also, docs repo will need update of https://github.com/Mellanox/network-operator-docs/blob/main/templates/helm.rst.gotmpl

**Pull request merged**